core/src/main/java/com/linecorp/armeria/client/ClientFactoryBuilder.java

```diff
@@ -224,20 +227,30 @@ public ClientFactoryBuilder connectionPoolListener(
     }
 
     /**
+     * Sets the {@link MeterRegistry} which collects various stats.
+     */
+    public ClientFactoryBuilder meterRegistry(MeterRegistry meterRegistry) {
+        this.meterRegistry = requireNonNull(meterRegistry, "meterRegistry");
+        return this;
+    }
+
+    /**
      * Returns a newly-created {@link ClientFactory} based on the properties of this builder.
      */
     public ClientFactory build() {
+        final MeterRegistry meterRegistry = this.meterRegistry != null ? this.meterRegistry
```

Comment:
I'm fine with using `Optional` but it seems like it's not recommended to use `Optional` as a parameter or a member field according to IntelliJ's inspection. Let me leave this as it is for now.